plugins/flutter_aepedge/android/src/main/java/com/adobe/marketing/mobile/flutter/flutter_aepedge/FlutterAEPEdgeDataBridge.java

```diff
@@ -50,23 +52,36 @@ protected static ExperienceEvent eventFromMap(final Map map) {
              return null;
          }
 
-        Map<String, Object> xdmData = getNullableMap(map, XDM_DATA_KEY);
         String datasetId = null;
+        String datastreamIdOverride = null;
 
+        Map<String, Object> xdmData = getNullableMap(map, XDM_DATA_KEY);
 
-        if (xdmData != null) {
 
+        if (xdmData != null) {
             Map<String, Object> data = getNullableMap(map, DATA_KEY);
-
-            datasetId = getNullableString(map, DATASET_IDENTIFIER_KEY);
-            
-            ExperienceEvent event = new ExperienceEvent.Builder().setXdmSchema(xdmData, datasetId).setData(data).build();
-
+            Map<String, Object> datastreamConfigOverride = getNullableMap(map, DATASTREAM_CONFIG_OVERRIDE_KEY);
+
+
+            try {
+                datasetId = getNullableString(map, DATASET_IDENTIFIER_KEY);
+                datastreamIdOverride = getNullableString(map, DATASTREAM_ID_OVERRIDE_KEY);
+            } catch (Exception e) {
+                Log.d(TAG, "experienceEventFromReadableMap: " + e);
+            }
+
+            ExperienceEvent event;
+            if (datastreamIdOverride != null || datastreamConfigOverride != null) {
```

Comment:
Need to add multiple constructors on the flutter side in case Override and dataset Id both are present. Currently we just ignore the dataset id and use the overrides.

Having separate constructors would clarify the exclusive usages of datasetId and overrides.

Reply:
Updated in dart.